pandas/core/algorithms.py

```diff
@@ -198,8 +200,16 @@ def _reconstruct_data(
     -------
     ExtensionArray or np.ndarray
     """
+    if isinstance(values, ABCExtensionArray) and values.dtype == dtype:
+        # Catch DatetimeArray/TimedeltaArray
+        return values
+
     if is_extension_array_dtype(dtype):
-        values = dtype.construct_array_type()._from_sequence(values)
+        cls = dtype.construct_array_type()
```

Comment:
elsif here (why are using isintance when we should be using is_extension_array_dtype)?

Reply:
For the elif part of the comment, I think that Brock maybe keeping the early return separate for readability

Reply:
> elsif here

sure

> why are using isintance when we should be using is_extension_array_dtype

1) perf, is_foo_dtype is slow
2) to catch DTA/TDA

(... i think.  most of this is a resurrected branch from a ways back)